include/oneapi/dpl/pstl/algorithm_impl.h

```diff
@@ -1021,9 +1023,9 @@ __pattern_search_n(__parallel_tag<_IsVector> __tag, _ExecutionPolicy&& __exec, _
 {
     if (static_cast<_Size>(__last - __first) == __count)
     {
-        const bool __result =
-            !__internal::__pattern_any_of(__tag, ::std::forward<_ExecutionPolicy>(__exec), __first, __last,
-                                          [&__value, &__pred](const _Tp& __val) { return !__pred(__val, __value); });
+        const bool __result = !__internal::__pattern_any_of(__tag, std::forward<_ExecutionPolicy>(__exec), __first,
+            __last, [&__value, __pred](auto&& __val) mutable { return !__pred(std::forward<decltype(__val)>(__val),
+            __value); });
```

Comment:
Same comment here, why is `__value` captured by reference?

Reply:
The same answer.